### PR TITLE
CASMPET-2610: Add troubleshooting doc for Envoy SDS failure

### PR DESCRIPTION
**Pull request by brantk-hpe**
This adds a troubleshooting guide for a problem that we've seen
several times recently where the Envoy container disallows
outbound requests with an error like
`TLS error: Secret is not supplied by SDS`.

### CODE REVIEW

**Comment by brantk-hpe** [comment]
I need to backport this to 1.2, 1.1, and 1.0.

**Review by bo-quan — APPROVED** [review]

**Comment by mtupitsyn** [comment]
> I need to backport this to 1.2, 1.1, and 1.0.

@brantk-hpe To achieve that, you can add a comment which says `/backport release/1.0 release/1.1 release/1.2` to this PR. No need to wait for merge/approval of the PR. If change is backported and new changes are added to original PR, you can cleanup (i.e. delete backport branch) and re-backport.

**Pull request merged**
